Dockerfile

```diff
@@ -34,6 +34,20 @@ RUN hugo --gc --minify -d /out -e $HUGO_ENV -b $DOCS_URL
 FROM scratch as release
 COPY --from=build /out /
 
+FROM scratch as update-stats
+COPY --from=build /src/hugo_stats.json /hugo_stats.json
+
+FROM build as validate-stats
+RUN git diff -- hugo_stats.json > diff.txt
+COPY <<-"EOF" validate.awk
+$1 == "+++" { a = 1 }
+a == 1 && $1 == "+" {
+  print "ERROR: hugo_stats.json differs. Update with `docker buildx bake update-stats`";
+  exit 1
+}
+EOF
+RUN awk -f validate.awk diff.txt
```

Comment:
first I used an approach of just checking `git status --porcelain` but the problem with that was that depending on how you run `hugo` to build, `hugo_stats.json` will be different (`dev` build contains fewer lines than `prod`).

this awk script checks that the `dev` hugo_stats.json generated in `FROM build` doesn't contain any uncomitted lines.

Reply:
the more I look at this the more involved it seems. It's probably better to just enforce a porcelain check with the dev mode file. If you accidentally commit the prod file, the ci will fail. Should be fine

Reply:
`update-stats` bake target generates the correct one

Reply:
changed it so there's no discrepancy between dev and prod buildStats